### PR TITLE
Make SDK more resilient. Fixes GH-22

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,10 +2,11 @@ module github.com/netfoundry/ziti-sdk-golang
 
 go 1.14
 
-//replace github.com/netfoundry/ziti-foundation => ../ziti-foundation
+// replace github.com/netfoundry/ziti-foundation => ../ziti-foundation
 
 require (
 	github.com/Jeffail/gabs v1.4.0
+	github.com/cenkalti/backoff/v4 v4.0.2
```
Comment (ekoby) [inline go.mod:9]: I didn't find where this was used

File: ziti/ziti.go [hunk 2]
```diff
@@ -553,18 +639,261 @@ func (context *contextImpl) Close() {
...
+	}
+}
+
+func (mgr *listenerManager) createSessionWithBackoff() {
+	expBackoff := backoff.NewExponentialBackOff()
+	expBackoff.InitialInterval = 50 * time.Millisecond
+	expBackoff.MaxInterval = 10 * time.Second
+	expBackoff.MaxElapsedTime = mgr.options.GetConnectTimeout()
+
+	_ = backoff.Retry(mgr.createSession, expBackoff)
```
Comment (plorenz) [inline ziti/ziti.go:839]: new backoff library is used here

File: go.mod (hunk 1 above)
Comment (plorenz) [reply go.mod:9]: I put comments on the two places it is used

Comment (ekoby) [reply go.mod:9]: thanks, ziti.go was collapsed in the view so my search returned nothing

**Pull request merged**